packages/cli/src/models/network/NetworkController.ts

```diff
@@ -416,240 +424,240 @@ export default class NetworkController {
+  public async deployDependencyIfNeeded(depName: string, depVersion: string): Promise<void | never> {
     try {
-      const dependency = new Dependency(depName, depVersion)
-      if (dependency.isDeployedOnNetwork(this.network) || this.networkFile.dependencyHasMatchingCustomDeploy(depName)) return
-      log.info(`Deploying ${depName} contracts`)
-      const deployment = await dependency.deploy(this.txParams)
-      this.networkFile.setDependency(depName, { 
+      const dependency = new Dependency(depName, depVersion);
+      if (dependency.isDeployedOnNetwork(this.network) || this.networkFile.dependencyHasMatchingCustomDeploy(depName)) return;
+      log.info(`Deploying ${depName} contracts`);
+      const deployment = await dependency.deploy(this.txParams);
+      this.networkFile.setDependency(depName, {
         package: (await deployment.getProjectPackage()).address,
         version: deployment.version,
         customDeploy: true
-      })
+      });
     } catch (error) {
-      error.message = `Failed deployment of dependency ${depName} with error: ${error.message}`
-      throw error
+      error.message = `Failed deployment of dependency ${depName} with error: ${error.message}`;
+      throw error;
     }
   }
 
-  //DependencyController
-  async handleDependenciesLink() {
+  // DependencyController
+  public async handleDependenciesLink(): Promise<void> {
     await allPromisesOrError(_.concat(
       _.map(this.packageFile.dependencies, (version, dep) => this.linkDependency(dep, version)),
-      _.map(this.networkFile.dependenciesNamesMissingFromPackage(), dep => this.unlinkDependency(dep))
-    ))
+      _.map(this.networkFile.dependenciesNamesMissingFromPackage(), (dep) => this.unlinkDependency(dep))
+    ));
   }
 
-  //DependencyController
-  async unlinkDependency(depName) {
+  // DependencyController
+  public async unlinkDependency(depName: string): Promise<void | never> {
     try {
       if (await this.project.hasDependency(depName)) {
         log.info(`Unlinking dependency ${depName}`);
         await this.project.unsetDependency(depName);
       }
       this.networkFile.unsetDependency(depName);
     } catch (error) {
-      throw Error(`Failed to unlink dependency ${depName} with error: ${error.message}`)
+      throw Error(`Failed to unlink dependency ${depName} with error: ${error.message}`);
     }
   }
 
-  //DependencyController
-  async linkDependency(depName, depVersion) {
+  // DependencyController
+  public async linkDependency(depName: string, depVersion: string): Promise<boolean | never> {
     try {
       if (this.networkFile.dependencyHasMatchingCustomDeploy(depName)) {
         log.info(`Using custom deployment of ${depName}`);
         const depInfo = this.networkFile.getDependency(depName);
-        return await this.project.setDependency(depName, depInfo.package, depInfo.version);
+        return await (<AppProject>this.project).setDependency(depName, depInfo.package, depInfo.version);
```

Comment:
Mmmh. Having dependencies linked is possible in a `SimpleProject`. I'm not sure if this cast is accurate.

Reply:
Wow! Good catch =D